include/boost/url/const_string.hpp

```diff
@@ -0,0 +1,216 @@
+        Allocator const& a,
+        char*& dest);
+
+public:
+    /** The default allocator type to construct read-only instances of @ref const_string
+
+        An allocator of this class can be instantiated to create
+        multiple read-only string values.
+
+        @note This custom allocator is not an instance of the C++ Allocator named requirement
+
+        @par References
+        @li <a href="https://en.cppreference.com/w/cpp/named_req/Allocator">Allocator (cppreference)</a>
+
+     */
+    class allocator;
+
+    /** Destructor
+
+        Upon destruction, ownership of the shared string is released. If
+        there are no more owners, the resources for the string are released.
+
+     */
+    inline
+    ~const_string();
+
+    /** Default constructor
+
+        This function creates the default base @ref string_view, which points to
+        no string / nullptr.
+
+        @note As an optimization, the reference counter is `nullptr`.
+
+     */
+    const_string() = default;
+
+    /** Construct an initial const_string
+
+        This function creates a string of the specified size at the
+        destination with the a custom allocator.
+
+        @note Although this function returns sets the destination pointer
+        to the new string address, this const_string owns the buffer.
+        The buffer is destroyed after the last const_string points to it.
+
+        @tparam Allocator Allocator type
+
+        @param n Number of chars in the string `dest`
+
+        @param a Allocator used to create a copy of `dest`
+
+        @param dest Destination
+
+     */
+    template<class Allocator>
+    const_string(
+        std::size_t n,
+        Allocator const& a,
+        char*& dest);
+
+    /** Construct a copy of a @ref const_string
+
+        This function constructs a copy of a @ref const_string with the
+        specified allocator.
+
+        This function allocates a @ref const_string the specified allocator
+        and copies its contents to this new string.
+
+        @tparam Allocator Allocator type
+
+        @param s string_view whose contents will be copied to this const_string
+
+        @param a allocator used to create this const_string
+
+     */
+    template< class Allocator =
+        std::allocator<char> >
+    explicit const_string(
+        string_view s,
+        Allocator const& a = {});
+
+    /** Copy constructor
+
+        This function constructs a copy of a const_string. As const_string
+        is read-only, this function can only copy and increment the string
+        reference counter.
+
+        @param other The other const_string
+
+     */
+    inline const_string(const_string const& other) noexcept;
+
+    /** Copy assignment
+
+        This function copy assigns a @ref const_string.
+
+        @note As @ref const_string is read-only, this function can only copy and increment
+        the string reference counter, which allows this operation to execute in constant time.
```

Comment:
You don't need to say anything about being read-only or reference counting. Instead you could just wrote
```
/** Copy assignment

    Objects of this type are cheap to copy.
    
    @par Complexity
    Constant time.
    ...
```